rotkehlchen/tests/api/test_current_assets_price.py

```diff
@@ -71,3 +81,226 @@ def test_get_current_assets_price_in_btc(rotkehlchen_api_server):
+
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'currentassetspriceresource',
+        ),
+        json={
+            'target_asset': A_EUR.identifier,
+            'assets': [A_ETH.identifier, A_USD.identifier],
+        },
+    )
+    result = assert_proper_response_with_result(response)
+    assert result['assets']['ETH'] == '45'
+    assert result['assets']['USD'] == '25'
+
+    # Now check that if there is a manual historical price entry at the same timestamp as our
+    # manual current price, proper error is returned.
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        manual_current_timestamp = cursor.execute(
+            'SELECT timestamp FROM price_history WHERE source_type=? AND from_asset=?',
+            (HistoricalPriceOracle.MANUAL_CURRENT.serialize_for_db(), A_ETH.identifier),
+        ).fetchone()[0]
+
+    GlobalDBHandler().add_single_historical_price(entry=HistoricalPrice(
+        from_asset=A_ETH,
+        to_asset=A_EUR,
+        timestamp=manual_current_timestamp,
+        price=ONE,
+        source=HistoricalPriceOracle.MANUAL,
+    ))
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'currentassetspriceresource',
+        ),
+        json={
+            'from_asset': A_ETH.identifier,
+            'to_asset': A_EUR.identifier,
+            'price': '78',
+        },
+    )
+    assert_error_response(
+        response=response,
+        contained_in_msg=f'already exists at {timestamp_to_date(ts=manual_current_timestamp)}',
+        status_code=HTTPStatus.CONFLICT,
+    )
+
+
+def test_remove_manual_current_price(rotkehlchen_api_server):
+    GlobalDBHandler().add_manual_current_price(
+        from_asset=A_ETH,
+        to_asset=A_EUR,
+        price=Price(FVal(10)),
+    )
+    GlobalDBHandler().add_manual_current_price(
+        from_asset=A_USD,
+        to_asset=A_EUR,
+        price=Price(FVal(25)),
+    )
+    response = requests.delete(
+        api_url_for(
+            rotkehlchen_api_server,
+            'currentassetspriceresource',
+        ),
+        json={
+            'asset': A_ETH.identifier,
+        },
+    )
+    assert_proper_response(response)
+    response = requests.delete(
+        api_url_for(
+            rotkehlchen_api_server,
+            'currentassetspriceresource',
+        ),
+        json={
+            'asset': A_ETH.identifier,
+        },
+    )
+    assert_error_response(
+        response=response,
+        contained_in_msg='Not found manual current price to delete',
+        status_code=HTTPStatus.CONFLICT,
+    )
+    # usd manual current price should have not been touched
+    assert GlobalDBHandler().get_manual_current_price(A_USD) == (A_EUR, Price(FVal(25)))
+
+
+@pytest.mark.parametrize('should_mock_current_price_queries', [False])
+def test_manual_current_prices_loop(inquirer):
+    GlobalDBHandler().add_manual_current_price(
```

Comment:
Having a description about what this test is for would be nice to have since we will forget in a few weeks